### ISSUE TITLE
Show loaded plugins source information(http or file)

### ISSUE THREAD

**Issue by neelayu** [influxdata/telegraf#16269]
### Use Case

In our Telegraf deployment, we run Telegraf using the `--config-directory` flag to load multiple configuration files. During initialization, Telegraf logs the counts of loaded inputs, outputs, aggregators, processors, and secretstores. However, with a large number of configuration files, it becomes challenging to trace which plugin is sourced from which file, making debugging and troubleshooting more difficult.

### Expected behavior

Show file name from where the plugin was loaded

### Actual behavior

Only file name and plugin count printed at startup.

### Additional info

To some extent it is closely related to #12827 and #15732

**Comment by Hipska** [comment]
I feel like this request is implemented in a wrong way.

To me this looks much more applicable for a separate command, rather than an option when starting/running telegraf. The command could look like `telegraf config source` or maybe `telegraf config check source`. As it seems there is currently no way to retrieve this table without actually starting the plugins, which feels like a shame.

What do you all think?

cc @mstrandboge @srebhan

**Comment by neelayu** [comment]
I get your point. But in your case, we still have to pass in the necessary flags `--config` and `--config-directory`, right?

**Comment by Hipska** [comment]
Yes of course! (unless you only want to load the default `/etc/telegraf/telegraf.conf`)
